maven-core/src/main/java/org/apache/maven/lifecycle/internal/MojoDescriptorCreator.java

```diff
@@ -176,18 +176,31 @@ public MojoDescriptor getMojoDescriptor( String task, MavenSession session, Mave
         }
         else if ( numTokens == 3 )
         {
-            // We have everything that we need except the version
-            //
-            // org.apache.maven.plugins:maven-remote-resources-plugin:???:process
-            //
-            // groupId
-            // artifactId
-            // ???
-            // goal
-            //
-            plugin = new Plugin();
-            plugin.setGroupId( tok.nextToken() );
-            plugin.setArtifactId( tok.nextToken() );
+            // groupId:artifactId:goal or pluginPrefix:version:goal (since Maven 3.9)
+
+            String firstToken = tok.nextToken();
+            // groupId or pluginPrefix? heuristics: groupId contains . but not pluginPrefix
```

Comment:
...contains dot (.), but ...

Reply:
yes, but what?

Reply:
It must be clear that this isn't a typo. That's why the symbol name and itself in parens.

Reply:
I don't understand the sentence, then I can't transform it into any action, sorry

Reply:
oh, I now understood...
this is the type of fixes where adding a commit is easier than commenting...

Reply:
Good